File: packages/schemas/src/browser.ts
export const BROWSER_PROTOCOL_VERSION = 1 as const;

export type BrowserContextMode = "selection" | "main" | "visible";
export type BrowserConnectionState = "disconnected" | "connecting" | "connected" | "error";
export type BrowserRequestStatus = "completed" | "permission_required" | "unavailable" | "failed";
export type BrowserPermissionKind = "temporary_active_tab" | "optional_origin";
export type BrowserAuditDecision = "allowed" | "blocked" | "failed";
export type BrowserFieldKind = "textarea" | "text_input" | "contenteditable";
export type BrowserAdapterHealthState = "available" | "degraded" | "unsupported";
export type BrowserDraftTone = "reply" | "regenerate" | "shorten" | "formalize";
export type BrowserActionKind = "open_tab" | "fill_field" | "click";
export type BrowserActionPlanStatus =
  | "pending_confirmation"
  | "confirmed"
  | "policy_authorized"
  | "executing"
  | "completed"
  | "cancelled"
  | "expired"
  | "failed"
  | "blocked";
export type WhatsAppBusyDecision = "allowed" | "blocked" | "requires_confirmation";
export type WhatsAppBusyCategory =
  | "normal"
  | "urgent"
  | "otp"
  | "password"
  | "payment"
  | "legal"
  | "medical"
  | "security"
  | "unknown";
export type BrowserPersonalityPreset = "concise" | "supportive" | "playful" | "professional" | "custom";

export interface BrowserLandmark {
  role: string;
  name: string;
  text: string;
}

export interface BrowserAvailableAction {
  actionId: string;
  capability: "page.read" | "tab.open" | "field.detect" | "field.fill" | "message.draft_reply" | "message.send_reply";
  label: string;
}

export interface BrowserAdapterCapability {
  id: "page.read" | "tab.open" | "field.detect" | "field.fill" | "message.draft_reply" | "message.send_reply";
  label: string;
  available: boolean;
  detail: string;
}

export interface BrowserAdapterHealth {
  adapterId: string;
  adapterVersion: number;
  state: BrowserAdapterHealthState;
  detail: string;
  capabilities: BrowserAdapterCapability[];
}

export interface BrowserWritableField {
  handle: string;
  kind: BrowserFieldKind;
  label: string;
  placeholder?: string | null;
  valuePreview: string;
  valueCharacterCount: number;
  maxLength?: number | null;
  required: boolean;
  disabled: boolean;
  capturedAt: string;
}

export interface BrowserPageContext {
  pageSessionId: string;
  origin: string;
  url: string;
  title: string;
  adapterId: string;
  adapterVersion: number;
  mode: BrowserContextMode;
  capturedAt: string;
  visibleText: string;
  selectionText?: string | null;
  mainText?: string | null;
  landmarks: BrowserLandmark[];
  availableActions: BrowserAvailableAction[];
  writableFields: BrowserWritableField[];
  adapterHealth: BrowserAdapterHealth;
  characterCount: number;
  truncated: boolean;
}

export interface BrowserSession {
  id: string;
  origin: string;
  url: string;
  title: string;
  adapterId: string;
  mode: BrowserContextMode;
  characterCount: number;
  truncated: boolean;
  createdAt: string;
  updatedAt: string;
  expiresAt: string;
}

export interface BrowserSessionListResponse {
  items: BrowserSession[];
  total: number;
}

export interface BrowserPermission {
  origin: string;
  kind: BrowserPermissionKind;
  granted: boolean;
  grantedAt?: string | null;
  detail: string;
}

export interface BrowserPermissionListResponse {
  items: BrowserPermission[];
  total: number;
}

export interface BrowserPermissionRequest {
  origin?: string | null;
  kind?: BrowserPermissionKind;
}

export interface BrowserPermissionResponse {
  status: BrowserRequestStatus;
  permission?: BrowserPermission | null;
  instruction: string;
}

export interface BrowserStatusResponse {
  state: BrowserConnectionState;
  connected: boolean;
  browserName?: string | null;
  browserVersion?: string | null;
  extensionVersion?: string | null;
  extensionOrigin?: string | null;
  protocolVersion: number;
  activeSessions: number;
  permissions: number;
  lastConnectedAt?: string | null;
  lastMessageAt?: string | null;
  lastError?: string | null;
  credentialPath: string;
}

export interface BrowserContextReadRequest {
  mode?: BrowserContextMode;
  userApproved?: boolean;
}

export interface BrowserContextReadResponse {
  status: BrowserRequestStatus;
  context?: BrowserPageContext | null;
  instruction?: string | null;
}

export interface BrowserContextSummaryRequest {
  mode?: BrowserContextMode;
  instruction?: string;
  userApproved?: boolean;
}

export interface BrowserContextSummaryResponse {
  status: BrowserRequestStatus;
  summary: string;
  model?: string | null;
  latencyMs: number;
  context?: BrowserPageContext | null;
  instruction?: string | null;
}

export interface BrowserSearchRequest {
  query: string;
  limit?: number;
  userApproved?: boolean;
}

export interface BrowserSearchItem {
  title: string;
  summary: string;
  url?: string | null;
  source?: string | null;
}

export interface BrowserSearchResponse {
  status: BrowserRequestStatus;
  query: string;
  items: BrowserSearchItem[];
  summary: string;
}

export interface BrowserOpenTabRequest {
  url: string;
  active?: boolean;
  userApproved?: boolean;
}

export interface BrowserOpenTabResponse {
  status: BrowserRequestStatus;
  url: string;
  tabId?: number | null;
  instruction?: string | null;
}

export interface BrowserDraftReplyRequest {
  instruction: string;
  mode?: BrowserContextMode;
  pageSessionId?: string | null;
  fieldHandle?: string | null;
  tone?: BrowserDraftTone;
  userApproved?: boolean;
}

export interface BrowserDraftReplyResponse {
  status: BrowserRequestStatus;
  draft: string;
  field?: BrowserWritableField | null;
  context?: BrowserPageContext | null;
  model?: string | null;
  latencyMs: number;
  instruction?: string | null;
}

export interface BrowserFillFieldRequest {
  pageSessionId: string;
  fieldHandle: string;
  value: string;
  userApproved?: boolean;
}

export interface BrowserFillFieldResponse {
  status: BrowserRequestStatus;
  fieldHandle: string;
  inserted: boolean;
  previousValuePreview?: string | null;
  instruction?: string | null;
}

export interface BrowserClearFieldRequest {
  pageSessionId: string;
  fieldHandle: string;
  restoreOriginal?: boolean;
  userApproved?: boolean;
}

export interface BrowserActionStep {
  id: string;
  kind: BrowserActionKind;
  label: string;
  origin?: string | null;
  pageSessionId?: string | null;
  fieldHandle?: string | null;
  url?: string | null;
  value?: string | null;
  actionId?: string | null;
  targetLabel?: string | null;
  verification: string;
}

export interface BrowserActionPlan {
  id: string;
  status: BrowserActionPlanStatus;
  summary: string;
  previewMarkdown: string;
  origin?: string | null;
  pageSessionId?: string | null;
  steps: BrowserActionStep[];
  confirmationToken?: string | null;
  expiresAt: string;
  createdAt: string;
  updatedAt: string;
  resultDetail?: string | null;
}

export interface BrowserActionPlanCreateRequest {
  chain: "open_url" | "fill_field" | "whatsapp_send";
  url?: string | null;
  pageSessionId?: string | null;
  fieldHandle?: string | null;
  value?: string | null;
  targetLabel?: string | null;
  userApproved?: boolean;
}

export interface BrowserClickActionRequest {
  pageSessionId: string;
  actionId: string;
  expectedText?: string | null;
  targetLabel?: string | null;
  userApproved?: boolean;
}

export interface BrowserClickActionResponse {
  status: BrowserRequestStatus;
  actionId: string;
  clicked: boolean;
  verified: boolean;
  instruction?: string | null;
}

export interface BrowserActionPlanResponse {
  status: BrowserRequestStatus;
  plan?: BrowserActionPlan | null;
  instruction?: string | null;
}

export interface BrowserActionPlanListResponse {
  items: BrowserActionPlan[];
  total: number;
}

export interface BrowserActionConfirmRequest {
  planId: string;
  confirmationToken: string;
}

export interface BrowserEmergencyStopResponse {
  stopped: boolean;
  cancelledPlans: number;
  instruction: string;
}

export interface WhatsAppBusyModePolicy {
  enabled: boolean;
  allowlistedContacts: string[];
  allowGroups: boolean;
  timezone: string;
  windowStart: string;
  windowEnd: string;
  cooldownMinutes: number;
  dailyLimit: number;
  template: string;
  emergencyStopped: boolean;
  permissionOrigin: string;
  permissionGranted: boolean;
  updatedAt: string;
}

export interface WhatsAppBusyModePolicyPatch {
  enabled?: boolean;
  allowlistedContacts?: string[];
  allowGroups?: boolean;
  timezone?: string;
  windowStart?: string;
  windowEnd?: string;
  cooldownMinutes?: number;
  dailyLimit?: number;
  template?: string;
  resetEmergencyStop?: boolean;
}

export interface WhatsAppBusyModePolicyResponse {
  status: BrowserRequestStatus;
  policy: WhatsAppBusyModePolicy;
  permission?: BrowserPermissionResponse | null;
  instruction?: string | null;
}

export interface WhatsAppBusyModeEvaluationRequest {
  pageSessionId?: string | null;
  contactLabel?: string | null;
  isGroup?: boolean | null;
  latestMessageText?: string | null;
  userApproved?: boolean;
}

export interface WhatsAppBusyModeEvaluationResponse {
  status: BrowserRequestStatus;
  allowed: boolean;
  decision: WhatsAppBusyDecision;
  reason: string;
  category: WhatsAppBusyCategory;
  urgencyDetected: boolean;
  ownerNotification: boolean;
  targetLabel?: string | null;
  draft?: string | null;
  policy: WhatsAppBusyModePolicy;
}

export interface WhatsAppBusyModeSendRequest {
  pageSessionId: string;
  fieldHandle?: string | null;
  latestMessageText?: string | null;
  userApproved?: boolean;
}

export interface WhatsAppBusyModeSendResponse {
  status: BrowserRequestStatus;
  evaluation: WhatsAppBusyModeEvaluationResponse;
  plan?: BrowserActionPlan | null;
  instruction?: string | null;
}

export interface BrowserPersonalityProfile {
  preset: BrowserPersonalityPreset;
  displayName: string;
  customInstruction: string;
  writerTemperature: number;
  maxDraftCharacters: number;
  automationDisclosure: string;
  updatedAt: string;
}

export interface BrowserPersonalityProfilePatch {
  preset?: BrowserPersonalityPreset;
  displayName?: string;
  customInstruction?: string;
  writerTemperature?: number;
  maxDraftCharacters?: number;
  automationDisclosure?: string;
}

export interface BrowserContactTonePreference {
  adapterId: string;
  contactLabel: string;
  toneInstruction: string;
  approved: boolean;
  updatedAt: string;
}

export interface BrowserContactTonePreferenceRequest {
  adapterId: string;
  contactLabel: string;
  toneInstruction: string;
  approved?: boolean;
}

export interface BrowserMoodHint {
  label: string;
  confidence: number;
  expiresAt: string;
  persisted: boolean;
}

export interface BrowserMoodInferRequest {
  text: string;
  ttlSeconds?: number;
}

export interface BrowserPersonalitySettingsResponse {
  profile: BrowserPersonalityProfile;
  contactTones: BrowserContactTonePreference[];
  moodHint?: BrowserMoodHint | null;
}

export interface BrowserPersonalityPreviewRequest {
  sampleText?: string;
  contactLabel?: string | null;
  adapterId?: string | null;
}

export interface BrowserPersonalityPreviewResponse {
  preview: string;
  profile: BrowserPersonalityProfile;
  contactTone?: BrowserContactTonePreference | null;
  moodHint?: BrowserMoodHint | null;
}

export type BrowserVoiceIntent = "read_page" | "summarize_page" | "draft_reply" | "search_web" | "open_url" | "unknown";

export interface BrowserVoiceCommandRequest {
  transcript: string;
  mode?: BrowserContextMode;
  pageSessionId?: string | null;
  userApproved?: boolean;
}

export interface BrowserVoiceCommandResponse {
  status: BrowserRequestStatus;
  transcriptPreview: string;
  intent: BrowserVoiceIntent;
  instruction: string;
  summary?: BrowserContextSummaryResponse | null;
  draft?: BrowserDraftReplyResponse | null;
  search?: BrowserSearchResponse | null;
  actionPlan?: BrowserActionPlan | null;
}

export interface BrowserDisconnectResponse {
  disconnected: boolean;
  pageSessionId: string;
}

export interface BrowserAuditEvent {
  id: string;
  eventType: string;
  decision: BrowserAuditDecision;
  operation: string;
  origin?: string | null;
  pageSessionId?: string | null;
  detail: string;
  payloadSha256?: string | null;
  payloadCharacterCount: number;
  createdAt: string;
}

export interface BrowserAuditListResponse {
  items: BrowserAuditEvent[];
  total: number;
}

export interface BrowserBridgeEnvelope<TPayload = Record<string, unknown>> {
  protocolVersion: number;
  requestId: string;
  type: string;
  timestamp: string;
  pageSessionId?: string | null;
  payload: TPayload;
}

export interface BrowserBridgeReadyPayload {
  browserName: string;
  browserVersion?: string | null;
  extensionVersion: string;
  permissions: BrowserPermission[];
}

export type BrowserConnectionChangedEvent = {
  id: string;
  type: "browser.connection.changed";
  timestamp: string;
  payload: BrowserStatusResponse;
};

export type BrowserPageContextUpdatedEvent = {
  id: string;
  type: "browser.page.context.updated";
  timestamp: string;
  payload: {
    context: BrowserPageContext;
    session: BrowserSession;
  };
};

export type BrowserPageSessionClosedEvent = {
  id: string;
  type: "browser.page.session.closed";
  timestamp: string;
  payload: { pageSessionId: string };
};

export type BrowserPermissionChangedEvent = {
  id: string;
  type: "browser.permission.changed";
  timestamp: string;
  payload: { permissions: BrowserPermission[] };
};

export type BrowserCoreEvent =
  | BrowserConnectionChangedEvent
  | BrowserPageContextUpdatedEvent
  | BrowserPageSessionClosedEvent
  | BrowserPermissionChangedEvent;

File: services/core/src/deyana_core/api/browser.py
from __future__ import annotations

from fastapi import APIRouter, HTTPException, Query, Request

from ..browser.models import (
    BrowserActionConfirmRequest,
    BrowserActionPlanCreateRequest,
    BrowserActionPlanListResponse,
    BrowserActionPlanResponse,
    BrowserEmergencyStopResponse,
    BrowserAuditListResponse,
    BrowserContextReadRequest,
    BrowserContextReadResponse,
    BrowserContextSummaryRequest,
    BrowserContextSummaryResponse,
    BrowserClearFieldRequest,
    BrowserClickActionRequest,
    BrowserClickActionResponse,
    BrowserContactTonePreference,
    BrowserContactTonePreferenceRequest,
    BrowserDisconnectResponse,
    BrowserDraftReplyRequest,
    BrowserDraftReplyResponse,
    BrowserFillFieldRequest,
    BrowserFillFieldResponse,
    BrowserMoodHint,
    BrowserMoodInferRequest,
    BrowserOpenTabRequest,
    BrowserOpenTabResponse,
    BrowserPersonalityPreviewRequest,
    BrowserPersonalityPreviewResponse,
    BrowserPersonalityProfile,
    BrowserPersonalityProfilePatch,
    BrowserPersonalitySettingsResponse,
    BrowserPermissionListResponse,
    BrowserPermissionRequest,
    BrowserPermissionResponse,
    BrowserSearchRequest,
    BrowserSearchResponse,
    BrowserSessionListResponse,
    BrowserStatusResponse,
    BrowserVoiceCommandRequest,
    BrowserVoiceCommandResponse,
    WhatsAppBusyModeEvaluationRequest,
    WhatsAppBusyModeEvaluationResponse,
    WhatsAppBusyModePolicy,
    WhatsAppBusyModePolicyPatch,
    WhatsAppBusyModePolicyResponse,
    WhatsAppBusyModeSendRequest,
    WhatsAppBusyModeSendResponse,
)
from ..privacy import PrivacyPolicyError
from ..tools import ToolExecutionError

router = APIRouter(prefix="/browser", tags=["browser"])


@router.get("/status", response_model=BrowserStatusResponse)
async def browser_status(request: Request) -> BrowserStatusResponse:
    return request.app.state.runtime.browser_service.status()


@router.get("/sessions", response_model=BrowserSessionListResponse)
async def browser_sessions(request: Request) -> BrowserSessionListResponse:
    return request.app.state.runtime.browser_service.list_sessions()


@router.delete("/sessions/{page_session_id}", response_model=BrowserDisconnectResponse)
async def disconnect_browser_session(
    request: Request,
    page_session_id: str,
) -> BrowserDisconnectResponse:
    return await request.app.state.runtime.browser_service.disconnect_session(page_session_id)


@router.get("/permissions", response_model=BrowserPermissionListResponse)
async def browser_permissions(request: Request) -> BrowserPermissionListResponse:
    return request.app.state.runtime.browser_service.list_permissions()


@router.post("/permissions/request", response_model=BrowserPermissionResponse)
async def request_browser_permission(
    request: Request,
    payload: BrowserPermissionRequest,
) -> BrowserPermissionResponse:
    return await request.app.state.runtime.browser_service.request_permission(payload)


@router.delete("/permissions", response_model=BrowserPermissionResponse)
async def revoke_browser_permission(
    request: Request,
    origin: str = Query(min_length=1, max_length=2048),
) -> BrowserPermissionResponse:
    return await request.app.state.runtime.browser_service.revoke_permission(origin)


@router.post("/context/read", response_model=BrowserContextReadResponse)
async def read_browser_context(
    request: Request,
    payload: BrowserContextReadRequest,
) -> BrowserContextReadResponse:
    return await request.app.state.runtime.browser_service.request_context(payload)


@router.post("/context/summarize", response_model=BrowserContextSummaryResponse)
async def summarize_browser_context(
    request: Request,
    payload: BrowserContextSummaryRequest,
) -> BrowserContextSummaryResponse:
    return await request.app.state.runtime.browser_service.summarize_context(payload)


@router.post("/search", response_model=BrowserSearchResponse)
async def browser_search(
    request: Request,
    payload: BrowserSearchRequest,
) -> BrowserSearchResponse:
    try:
        return await request.app.state.runtime.browser_service.search(payload)
    except PrivacyPolicyError as error:
        raise HTTPException(status_code=403, detail=error.response.reason) from error
    except ToolExecutionError as error:
        raise HTTPException(status_code=400, detail=str(error)) from error


@router.post("/tabs/open", response_model=BrowserOpenTabResponse)
async def open_browser_tab(
    request: Request,
    payload: BrowserOpenTabRequest,
) -> BrowserOpenTabResponse:
    try:
        return await request.app.state.runtime.browser_service.open_tab(payload)
    except ValueError as error:
        raise HTTPException(status_code=400, detail=str(error)) from error


@router.post("/messages/draft-reply", response_model=BrowserDraftReplyResponse)
async def draft_browser_reply(
    request: Request,
    payload: BrowserDraftReplyRequest,
) -> BrowserDraftReplyResponse:
    return await request.app.state.runtime.browser_service.draft_reply(payload)


@router.post("/fields/fill", response_model=BrowserFillFieldResponse)
async def fill_browser_field(
    request: Request,
    payload: BrowserFillFieldRequest,
) -> BrowserFillFieldResponse:
    return await request.app.state.runtime.browser_service.fill_field(payload)


@router.post("/fields/clear", response_model=BrowserFillFieldResponse)
async def clear_browser_field(
    request: Request,
    payload: BrowserClearFieldRequest,
) -> BrowserFillFieldResponse:
    return await request.app.state.runtime.browser_service.clear_field(payload)


@router.post("/actions/click", response_model=BrowserClickActionResponse)
async def click_browser_action(
    request: Request,
    payload: BrowserClickActionRequest,
) -> BrowserClickActionResponse:
    return await request.app.state.runtime.browser_service.click_action(payload)


@router.get("/audit", response_model=BrowserAuditListResponse)
async def browser_audit(
    request: Request,
    limit: int = Query(default=50, ge=1, le=200),
) -> BrowserAuditListResponse:
    return request.app.state.runtime.browser_service.list_audit(limit=limit)


@router.get("/actions/plans", response_model=BrowserActionPlanListResponse)
async def browser_action_plans(
    request: Request,
    limit: int = Query(default=20, ge=1, le=100),
) -> BrowserActionPlanListResponse:
    return request.app.state.runtime.browser_service.list_action_plans(limit=limit)


@router.post("/actions/plans", response_model=BrowserActionPlanResponse)
async def create_browser_action_plan(
    request: Request,
    payload: BrowserActionPlanCreateRequest,
) -> BrowserActionPlanResponse:
    return request.app.state.runtime.browser_service.create_action_plan(payload)


@router.post("/actions/confirm", response_model=BrowserActionPlanResponse)
async def confirm_browser_action_plan(
    request: Request,
    payload: BrowserActionConfirmRequest,
) -> BrowserActionPlanResponse:
    return request.app.state.runtime.browser_service.confirm_action_plan(payload)


@router.post("/actions/plans/{plan_id}/execute", response_model=BrowserActionPlanResponse)
async def execute_browser_action_plan(
    request: Request,
    plan_id: str,
) -> BrowserActionPlanResponse:
    return await request.app.state.runtime.browser_service.execute_action_plan(plan_id)


@router.post("/actions/plans/{plan_id}/cancel", response_model=BrowserActionPlanResponse)
async def cancel_browser_action_plan(
    request: Request,
    plan_id: str,
) -> BrowserActionPlanResponse:
    return request.app.state.runtime.browser_service.cancel_action_plan(plan_id)


@router.post("/actions/emergency-stop", response_model=BrowserEmergencyStopResponse)
async def browser_emergency_stop(request: Request) -> BrowserEmergencyStopResponse:
    return request.app.state.runtime.browser_service.emergency_stop()


@router.get("/whatsapp/busy-mode", response_model=WhatsAppBusyModePolicy)
async def get_whatsapp_busy_mode(request: Request) -> WhatsAppBusyModePolicy:
    return request.app.state.runtime.browser_service.get_whatsapp_busy_policy()


@router.patch("/whatsapp/busy-mode", response_model=WhatsAppBusyModePolicyResponse)
async def patch_whatsapp_busy_mode(
    request: Request,
    payload: WhatsAppBusyModePolicyPatch,
) -> WhatsAppBusyModePolicyResponse:
    try:
        return await request.app.state.runtime.browser_service.update_whatsapp_busy_policy(payload)
    except ValueError as error:
        raise HTTPException(status_code=400, detail=str(error)) from error


@router.post("/whatsapp/busy-mode/evaluate", response_model=WhatsAppBusyModeEvaluationResponse)
async def evaluate_whatsapp_busy_mode(
    request: Request,
    payload: WhatsAppBusyModeEvaluationRequest,
) -> WhatsAppBusyModeEvaluationResponse:
    return request.app.state.runtime.browser_service.evaluate_whatsapp_busy_mode(payload)


@router.post("/whatsapp/busy-mode/send", response_model=WhatsAppBusyModeSendResponse)
async def send_whatsapp_busy_reply(
    request: Request,
    payload: WhatsAppBusyModeSendRequest,
) -> WhatsAppBusyModeSendResponse:
    return await request.app.state.runtime.browser_service.send_whatsapp_busy_reply(payload)


@router.get("/personality", response_model=BrowserPersonalitySettingsResponse)
async def get_browser_personality(request: Request) -> BrowserPersonalitySettingsResponse:
    return request.app.state.runtime.browser_service.get_personality_settings()


@router.patch("/personality/profile", response_model=BrowserPersonalityProfile)
async def patch_browser_personality_profile(
    request: Request,
    payload: BrowserPersonalityProfilePatch,
) -> BrowserPersonalityProfile:
    try:
        return request.app.state.runtime.browser_service.update_personality_profile(payload)
    except ValueError as error:
        raise HTTPException(status_code=400, detail=str(error)) from error


@router.post("/personality/contact-tones", response_model=BrowserContactTonePreference)
async def save_browser_contact_tone(
    request: Request,
    payload: BrowserContactTonePreferenceRequest,
) -> BrowserContactTonePreference:
    return request.app.state.runtime.browser_service.save_contact_tone(payload)


@router.delete("/personality/contact-tones", response_model=dict[str, bool])
async def delete_browser_contact_tone(
    request: Request,
    adapter_id: str = Query(min_length=1, max_length=80, alias="adapterId"),
    contact_label: str = Query(min_length=1, max_length=160, alias="contactLabel"),
) -> dict[str, bool]:
    deleted = request.app.state.runtime.browser_service.delete_contact_tone(adapter_id, contact_label)
    return {"deleted": deleted}


@router.post("/personality/mood/infer", response_model=BrowserMoodHint)
async def infer_browser_mood(
    request: Request,
    payload: BrowserMoodInferRequest,
) -> BrowserMoodHint:
    return request.app.state.runtime.browser_service.infer_mood(payload)


@router.post("/personality/preview", response_model=BrowserPersonalityPreviewResponse)
async def preview_browser_personality(
    request: Request,
    payload: BrowserPersonalityPreviewRequest,
) -> BrowserPersonalityPreviewResponse:
    return request.app.state.runtime.browser_service.preview_personality(payload)


@router.post("/voice/command", response_model=BrowserVoiceCommandResponse)
async def route_browser_voice_command(
    request: Request,
    payload: BrowserVoiceCommandRequest,
) -> BrowserVoiceCommandResponse:
    return await request.app.state.runtime.browser_service.route_voice_command(payload)
